### DISCUSSION TITLE
Rendering to useFieldArray from GET call

### DISCUSSION THREAD

**Discussion by stranzhay** [react-hook-form/react-hook-form#3074] [Q&A]
I currently have a form that is either POST or PUT depending on return value if an id is found from GET call. If the form is blank meaning that no id was found, adding fields to `useFieldArray` works fine the way I have it. However, when I am trying to render/populate the form is there was an id found, the `useArrayField` only seems to populate the first row. The parent is `AddFormBody` which is watching the grades array name and consuming `Grades`
 component found here https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/AddFormBody.js . The array of object in question is Grades in this file https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/Grades.js . Attaching screenshots.
<img width="632" alt="Screen Shot 2020-10-03 at 8 21 52 PM" src="https://user-images.githubusercontent.com/16522636/95004388-76315d00-05b8-11eb-9f2f-ef7d536cbb2b.png">
<img width="922" alt="Screen Shot 2020-10-03 at 8 21 44 PM" src="https://user-images.githubusercontent.com/16522636/95004399-8d704a80-05b8-11eb-90df-8650036217e3.png">

**Comment by bluebill1049** [comment]
have you tried reset() at useEffect?

**Reply by stranzhay** [reply]
where are you recommending to try it ? here ? https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/Grades.js#L15-L27

**Reply by bluebill1049** [reply]
yes

**Reply by stranzhay** [reply]
I seem to have solved my issue with adding `const watchedFields = watch("grades", fields);

  useEffect(() => {
    console.log(fields);
    fields.map((grade, index) => {
      setValue(`grades[${index}].grade`, grade.grade);
      setValue(`grades[${index}].score`, grade.score);
      setValue(`grades[${index}].date`, grade.date);
      console.log(fields);
    });
  }, [fields]);`

**Reply by stranzhay** [reply]
so the fields are populating automatically with results from GET, but if i delete and try to update with PUT my values for `grades` are now missing. which was previously working 
<img width="373" alt="Screen Shot 2020-10-04 at 1 13 16 AM" src="https://user-images.githubusercontent.com/16522636/95007575-43e72600-05df-11eb-9810-ad22c37f9631.png">

**Reply by stranzhay** [reply]
the `fields` array shows that the element in question is being removed, however it doesn't seem like the `grades` name element exists to the parent component where the submit is happening. https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/AddFormBody.js#L86
